pkg/containerwatcher/v1/container_watcher.go

```diff
@@ -386,13 +454,55 @@ func CreateIGContainerWatcher(cfg config.Config, applicationProfileManager appli
 		httpWorkerChan:         make(chan *tracerhttptype.Event, 500000),
 
 		// cache
-		ruleBindingPodNotify: ruleBindingPodNotify,
-		timeBasedContainers:  mapset.NewSet[string](),
-		ruleManagedPods:      mapset.NewSet[string](),
-		runtime:              runtime,
+		ruleBindingPodNotify:         ruleBindingPodNotify,
+		timeBasedContainers:          mapset.NewSet[string](),
+		ruleManagedPods:              mapset.NewSet[string](),
+		runtime:                      runtime,
+		thirdPartyTracers:            mapset.NewSet[containerwatcher.CustomTracer](),
```

Comment:
why not a map[string]containerwatcher.CustomTracer with key = tracer.Name ?

Reply:
Why a map because of efficiency? I expect < 10 tracers.

Reply:
well then why a Set? you could just go with a slice too...
let's use the map and simplify the loop elsewhere

Reply:
I don't see the benefit as we will need to iterate over the values and start each tracer.
We can go with a slice if you wish.

Reply:
LOL